src/app/Routes.tsx

```diff
@@ -36,16 +40,16 @@ const Routes: FC = () => (
       <Route path="/count" element={<Authed element={<Count />} />} />
       <Route path="/auth" element={<Auth />} />
       <Route
-        path={'/narratives/:id/:obj/:ver'}
+        path={narrativeSelectedPath}
         element={<Authed element={<Navigator />} />}
       />
       <Route
         path={'/narratives/:category'}
         element={<Authed element={<Navigator />} />}
       />
       <Route
-        path={'/narratives/:category/:id/:obj/:ver'}
-        element={<Navigator />}
+        path={narrativeSelectedPathWithCategory}
+        element={<Authed element={<Navigator />} />}
       />
       <Route path="/narratives" element={<Authed element={<Navigator />} />} />
```

Comment:
My view is that `Routes.tsx` is simple enough at this point, if it becomes unwieldy then we can revisit nesting. I think even with collections it will be fine flat for now.